builtin/docker/builder.go

```diff
@@ -180,9 +188,18 @@ func (b *Builder) Build(
 		dockerfile = newPath
 	}
 
-	contextDir, relDockerfile, err := build.GetContextFromLocalDir(src.Path, dockerfile)
-	if err != nil {
-		return nil, status.Errorf(codes.FailedPrecondition, "unable to create Docker context: %s", err)
+	var contextDir string
+
+	if b.config.Context == "" {
+		contextDir, relDockerfile, err = build.GetContextFromLocalDir(src.Path, dockerfile)
+		if err != nil {
+			return nil, status.Errorf(codes.FailedPrecondition, "unable to create Docker context: %s", err)
+		}
```

Comment:
Since this is duplicated with the other branch of the if, could you pull it out and just have the `path` passed in be controlled by the presence of `b.config.Context`?